templates/server/resources/coredns_configmap.epp

```diff
@@ -0,0 +1,23 @@
+.:53 {
+  errors
+  health {
+    lameduck 5s
+  }
+  ready
+  kubernetes <%= $k8s::server::resources::coredns::cluster_domain %> in-addr.arpa ip6.arpa {
```

Comment:
can you please update the template to explicitly pass the variable into the template?

Reply:
and how do i manage the variable passing if i bring my own template with other variables?

Reply:
I see 3 ways:

1. Leave things as is
2. Pass another class parameter with template variables (`Hash[String, Any] $template_variables = {}`) or something like (I don't like it though)
3. Pass the configmap contents instead of a template path, so user can do whatever they really want in their code.

Reply:
initially i wanted to do 3. but didn't know how to get variables in there, i wanted to pass it through hiera, but then i have a more static config map 🤔

Reply:
okay, i will refactor.... i see my customer now has a k8s_cluster_domain fact, so i can build the complete corefile in hiera :-)